src/app/home-page/widget/entry-box/entry-box.component.ts

```diff
@@ -0,0 +1,121 @@
+import { debounceTime, takeUntil } from 'rxjs/operators';
+import { WorkflowService } from 'app/shared/state/workflow.service';
+import { WorkflowQuery } from 'app/shared/state/workflow.query';
+import { MyEntriesQuery } from 'app/shared/state/my-entries.query';
+import { SessionQuery } from 'app/shared/session/session.query';
+import { SessionService } from 'app/shared/session/session.service';
+import { MyWorkflowsService } from 'app/myworkflows/myworkflows.service';
+import { RegisterToolService } from 'app/container/register-tool/register-tool.service';
+import { Base } from 'app/shared/base';
+import { Dockstore } from 'app/shared/dockstore.model';
+
+@Component({
+  selector: 'app-entry-box',
+  templateUrl: './entry-box.component.html',
+  styleUrls: ['./entry-box.component.scss'],
+})
+export class EntryBoxComponent extends Base implements OnInit {
+  Dockstore = Dockstore;
+  @Input() entryType: EntryUpdateTime.EntryTypeEnum;
+  entryTypeLowerCase: string;
+  filterText: string;
+  listOfEntries: Array<EntryUpdateTime> = [];
+  totalEntries: number = 0;
+  firstLoad = true;
+  user: User;
+  helpLink: string;
+  allEntriesLink: string;
+  public isLoading = true;
+
+  constructor(
+    private registerToolService: RegisterToolService,
+    protected userQuery: UserQuery,
+    protected workflowService: WorkflowService,
+    protected workflowQuery: WorkflowQuery,
+    protected myWorkflowsService: MyWorkflowsService,
+    protected myEntriesQuery: MyEntriesQuery,
+    protected sessionQuery: SessionQuery,
+    protected sessionService: SessionService,
+    private usersService: UsersService
+  ) {
+    super();
+    this.user = this.userQuery.getValue().user;
+  }
+
+  ngOnInit(): void {
+    this.getMyEntries();
+
+    if (this.entryType) {
+      this.entryTypeLowerCase = this.entryType.toLowerCase();
+    }
+
+    //Get the links for the specified entryType
+    if (this.entryType === EntryUpdateTime.EntryTypeEnum.WORKFLOW) {
+      this.helpLink = Dockstore.DOCUMENTATION_URL + '/getting-started/dockstore-workflows.html';
+      this.allEntriesLink = '/my-workflows/';
+    } else if (this.entryType === EntryUpdateTime.EntryTypeEnum.TOOL) {
+      this.helpLink = Dockstore.DOCUMENTATION_URL + '/getting-started/dockstore-tools.html';
+      this.allEntriesLink = '/my-tools/';
+    } else if (this.entryType === EntryUpdateTime.EntryTypeEnum.SERVICE) {
+      this.helpLink = Dockstore.DOCUMENTATION_URL + '/getting-started/getting-started-with-services.html';
+      this.allEntriesLink = '/my-services/';
+    }
+  }
+
+  getMyEntries() {
+    this.usersService
+      .getUserEntries(null, this.filterText)
+      .pipe(debounceTime(750), takeUntil(this.ngUnsubscribe))
+      .subscribe((myEntries: Array<EntryUpdateTime>) => {
+        this.listOfEntries = [];
+        myEntries.forEach((entry: EntryUpdateTime) => {
+          if (this.entryType === entry.entryType && entry.entryType === EntryUpdateTime.EntryTypeEnum.WORKFLOW) {
```

Comment:
Do you need the the second part of this if condition? Can't you just do `if (this.entryType === entry.entryType)`? It would simplify the code a lot, and you wouldn't even need the filter I suggest above (although I have a personal preference for using the filter).

Looking at it again, I guess tools complicate the issue, but I think you could have a special case for tools.